packages/runtime-core/src/vnode.ts

```diff
@@ -507,6 +507,7 @@ function _createVNode(
     // <component :is="vnode"/>
     // #2078 make sure to merge refs during the clone instead of overwriting it
     const cloned = cloneVNode(type, props, true /* mergeRef: true */)
```

Comment:
I wonder if cloneVNode shouldn't copy the currentScopeId itself. Maybe it fails in other scenarios